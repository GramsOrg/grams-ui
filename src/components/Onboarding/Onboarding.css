.max-height {
    height: 100%;
}

.layout {
    max-width: 450px;
}

.ltr {
    text-align: left;
}

.rtl {
    text-align: left;
}