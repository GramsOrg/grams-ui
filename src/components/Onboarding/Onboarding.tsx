import React, { useState, useEffect } from 'react';

import { useTranslation } from 'react-i18next';

import { useDirection } from '../../';

import { Grid } from 'semantic-ui-react';

import SelectLocale from '../SelectLocale';
import Terms from '../Terms';
import ProfileOptions from '../ProfileOptions';

import '../../main.css';
import CreateProfile from '../CreateProfile';

export interface IOnboardingProps {

  /**
   * Toggles between light and dark modes
   */
  inverted?: boolean;

  /**
   * Optional locale for onboarding. Will prompt the user to select their locale if undefined
   */
  locale?: string;

  /**
   * Adjusts the styles to fit smaller screen sizes.
   */
  mobile?: boolean;

  /**
   * A callback function called when a profile is created.
   */
  onCreate?: (name: string, password: string, isDeveloper: boolean) => void;

  /**
   * A callback function called when the user opts to load a .grams file.
   */
  onImportBackup?: () => void;

  /**
   * A callback function called when the user opts to load their 24-word seed.
   */
  onImportSeed?: () => void;

  /**
   * A callback function called when the user opts to sync their ledger device.
   */
  onLedger?: () => void;
}

const defaultProps = {
  inverted: false
};

const Onboarding = (props: IOnboardingProps) => {

  const { t, i18n } = useTranslation();
  const direction = useDirection();

  const selectLocale = (
    <SelectLocale onSelect={() => setStep(1)} />
  );

  const terms = (
    <Terms
      size={props.mobile ? 'small' : 'medium'}
      content='TBD'
      onAccept={() => setStep(2)}
    />
  );

  const profileOptions = (
    <ProfileOptions
      onSelect={(option) => {
        switch(option) {
          case 'new':
            setStep(3);
            break;
          case 'backup':
            props.onImportBackup?.();
            break;
          case 'seed':
            props.onImportSeed?.();
            break;
          case 'ledger':
            props.onLedger?.();
            break;
          default:
            setStep(3);
            break;
        }
      }}
    />
  );

  const createProfile = (
    <CreateProfile
      onCreate={props.onCreate}
    />
  );

  const [step, setStep] = useState(0);
  const [journey, setJourney] = useState([
    selectLocale,
    terms,
    profileOptions,
    createProfile
  ]);

  useEffect(() => {
    if (props.locale) {
      i18n.changeLanguage(props.locale);
      setStep(1);
    }
  }, [props.locale]);

  return (
    <Grid centered className='max-height'>
      <Grid.Column verticalAlign='middle' className={`layout ${direction}`}>
        {journey[step]}
      </Grid.Column>
    </Grid>
  );
};

Onboarding.defaultProps = defaultProps;

export default Onboarding;
