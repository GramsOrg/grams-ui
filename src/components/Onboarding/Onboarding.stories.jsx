import React from 'react';

import { action } from "@storybook/addon-actions";

import Onboarding from './Onboarding.tsx';

export default {
  title: 'Flows/Onboarding',
  component: Onboarding,
  parameters: {
    componentSubtitle: 'An end-to-end onboarding flow for creating a Grams profile',
    storyshots: { disable: false },
  },
  argTypes: {
    inverted: {
      control: 'boolean'
    }
  }
};

const Template = ({inverted, ...props}) => {
  return (
    <Onboarding
      inverted={inverted}
      onCreate={action("Trigger onCreate()")}
      onImportBackup={action("Trigger onImportBackup()")}
      onImportSeed={action("Trigger onImportSeed()")}
      onLedger={action("Trigger onLedger()")}
      {...props}
    />
  );
};

export const Default = Template.bind({});
Default.args = {
  inverted: false
}
