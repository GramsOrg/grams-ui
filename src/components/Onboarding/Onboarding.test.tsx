/**
 * @jest-environment jsdom
 */

import React from 'react';
import { describe, expect, test } from '@jest/globals';
import { render } from '@testing-library/react';

import Onboarding from './Onboarding';

describe('Onboarding Component', () => {
  test('it should match the snapshot', () => {
    const { asFragment } = render(<Onboarding />);
    expect(asFragment()).toMatchSnapshot();
  });
});
