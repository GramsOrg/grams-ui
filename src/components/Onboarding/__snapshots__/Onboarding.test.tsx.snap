// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding Component it should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="ui centered grid max-height"
  >
    <div
      class="middle aligned column layout ltr"
    >
      <div
        class="ui list"
        role="list"
      >
        <div
          class="item mb-small"
          role="listitem"
        >
          <div
            aria-expanded="false"
            class="ui fluid selection dropdown"
            role="listbox"
            tabindex="0"
          >
            <div
              aria-atomic="true"
              aria-live="polite"
              class="divider default text"
              role="alert"
            >
              Select language
            </div>
            <i
              aria-hidden="true"
              class="dropdown icon"
            />
            <div
              class="menu transition"
            >
              <div
                aria-checked="false"
                aria-selected="true"
                class="selected item"
                role="option"
                style="pointer-events: all;"
              >
                <span
                  class="text"
                >
                  English
                </span>
              </div>
              <div
                aria-checked="false"
                aria-selected="false"
                class="item"
                role="option"
                style="pointer-events: all;"
              >
                <span
                  class="text"
                >
                  عربي
                </span>
              </div>
              <div
                aria-checked="false"
                aria-selected="false"
                class="item"
                role="option"
                style="pointer-events: all;"
              >
                <span
                  class="text"
                >
                  Deutsch
                </span>
              </div>
              <div
                aria-checked="false"
                aria-selected="false"
                class="item"
                role="option"
                style="pointer-events: all;"
              >
                <span
                  class="text"
                >
                  Français
                </span>
              </div>
              <div
                aria-checked="false"
                aria-selected="false"
                class="item"
                role="option"
                style="pointer-events: all;"
              >
                <span
                  class="text"
                >
                  Español
                </span>
              </div>
              <div
                aria-checked="false"
                aria-selected="false"
                class="item"
                role="option"
                style="pointer-events: all;"
              >
                <span
                  class="text"
                >
                  中文
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="item"
          role="listitem"
        >
          <button
            class="ui fluid primary disabled button"
            disabled=""
            tabindex="-1"
          >
            Select
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
